homeworks/07-mars-viewer/src/components/favorites/favorites.tsx

```diff
@@ -0,0 +1,29 @@
+import React, {ReactElement} from "react";
+import s from "./favorites.module.scss";
+import {useSelector} from "react-redux";
+import {selectFavorites, selectImages} from "../../store/store";
+import {IImage} from "../../store/sols";
+import Image from "../image/image";
+
+const Favorites: React.FC = () => {
+	const favorites = useSelector(selectFavorites);
+	const images = useSelector(selectImages);
+
+	const result: ReactElement[] = [];
+
+	Object.values(images).forEach((solImages: IImage[]) => {
+		solImages.forEach(image => {
+			if (favorites.includes(image.id)) result.push(<Image {...image} />)
```

Comment:
1. better map data only and use non-mutating methods (excluding places with immer like reducers), reduce + filter may be used in this case
2. `key` property is not being passed here